### PR TITLE
fix: error handling for preact-spa 422 code

### PR DESCRIPTION
**Pull request by misamu**
Implements error handling for examples preact-spa API calls handling all returned error codes

Flow handling changed to follow react-spa format

fix: package.json script build:preact 

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation within the code base (if appropriate).

## Further comments

[Issue 53](https://github.com/ory/elements/issues/53)

### CODE REVIEW

File: examples/preact-spa/src/register.tsx [hunk 1]
```diff
@@ -1,21 +1,65 @@
...
-      .then(({ data: flow }) => {
-        setFlow(flow)
+      .then(({ data: flow }) => setFlow(flow))
+      .catch(sdkErrorHandler)
+  }
+
+  // submit the registration form data to Ory
+  const submitFlow = (body: UpdateRegistrationFlowBody) => {
+    // something unexpected went wrong and the flow was not set
+    if (!flow) return setLocation("/signup", { replace: true })
```
Comment (Benehiko) [inline examples/preact-spa/src/register.tsx:38]:
```suggestion
    if (!flow) return setLocation("/registration", { replace: true })
```

this was changed to `registration`, right?

**Review by Benehiko — CHANGES_REQUESTED** [review]
This looks good, i think we can merge it soon :)

Just a small change required

File: examples/preact-spa/src/register.tsx (hunk 1 above)
Comment (misamu) [reply examples/preact-spa/src/register.tsx:38]: Yes, Changed to default value that Kratos has.

**Review by Benehiko — APPROVED** [review]
Very nice! :)

Thank you for taking the time to work on this, it is much appreciated :heart:

**Pull request merged**